src/brsTypes/components/RoSGNode.ts

```diff
@@ -7,22 +7,59 @@ import { Int32 } from "../Int32";
 import { RoAssociativeArray } from "./RoAssociativeArray";
 import { RoArray } from "./RoArray";
 import { AAMember } from "./RoAssociativeArray";
+import { Float } from "../Float";
+import { Call } from "../../parser/Expression";
+import { objectExpression } from "@babel/types";
 
 class Field {
-    constructor(readonly type: string, readonly alwaysNotify: boolean) {}
+    private type: string;
+    private value: BrsType;
+    private observers = new Array<BrsString>();
+
+    constructor(value: BrsType, alwaysNotify?: boolean) {
```

Comment:
It can be saved by the constructor if it has a scope setting (https://kendaleiv.com/typescript-constructor-assignment-public-and-private-keywords/).  This should probably be changed to "readonly alwaysNotify: boolean" so it's publicly accessible, but nothing can change it.

Reply:
I'm declaring the argument as private to use it later when the field changes. I've change the `set` implementation to have a default value for this.